test/test_hungry_wallet.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2018 Yegor Bugayenko
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the 'Software'), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+require 'minitest/autorun'
+require 'webmock/minitest'
+require 'tmpdir'
+require_relative 'test__helper'
+require_relative 'fake_home'
+
+require_relative '../lib/zold/json_page'
+require_relative '../lib/zold/wallet.rb'
+require_relative '../lib/zold/hungry_wallets.rb'
+# CachedWallets test.
+# Author:: Yegor Bugayenko (yegor256@gmail.com)
+# Copyright:: Copyright (c) 2018 Yegor Bugayenko
+# License:: MIT
+class TestHungryWallets < Zold::Test
+  def test_find_wallet
+    FakeHome.new(log: test_log).run do |home|
+      remotes = home.remotes
+      remotes.add('localhost', 4096)
+      json = home.create_wallet_json
+      id = Zold::JsonPage.new(json).to_hash['id']
+      stub_request(:get, "http://localhost:4096/wallet/#{id}").to_return(status: 200, body: json)
+      FileUtils.rm(["#{home.dir}/#{id}#{Zold::Wallet::EXT}", "#{home.dir}/#{id}#{Zold::Wallet::EXT}.lock"])
```

Comment:
@emilianodellacasa but in this case is just for the test because the task is exactly to pull wallet when it is not present, I don't know well the internals, but if I use a different id to pull, it will trigger an error at `patch`, saying I should join at least one wallet. So I just removed the physical file.